### PR TITLE
Refactor counts and checks

### PR DESCRIPTION
**Pull request by WalshyDev**
(Tested - Works!)

Things done:
- [x] Refactor code to allow for more expandability (more heads)
- [x] Refactor code to reduce duplicate lines and just more work
- [x] Improve the code overall
- [x] Fixed bug in /hl command
- [x] Bypass permission

### CODE REVIEW

File: src/main/java/dev/j3fftw/headlimiter/Utils.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        new ThreadFactoryBuilder().setNameFormat("HeadLimiter-pool-%d").build()
+    );
+
+    private static final Set<String> permissionNodes = new LinkedHashSet<>();
+    private static final String PERMISSION_PREFIX = "headlimiter.permission.";
+
+    private Utils() {}
+
+    protected static void loadPermissions() {
+        final ConfigurationSection sec = HeadLimiter.getInstance().getConfig().getConfigurationSection("permission");
```
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/headlimiter/Utils.java:40]: Rename this variable.

File: src/main/java/dev/j3fftw/headlimiter/Utils.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        if (sec != null) {
+            permissionNodes.addAll(sec.getKeys(false));
+        }
+    }
+
+    public static int getMaxHeads(@Nonnull Player player) {
+        if (permissionsEnabled() && !permissionNodes.isEmpty()) {
+            for (String node : permissionNodes) {
```
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/headlimiter/Utils.java:48]: Change it to permissionNode

**Review by J3fftw1 — CHANGES_REQUESTED** [review]
If you remove acronyms
Lets make them more accurate aswell

File: src/main/java/dev/j3fftw/headlimiter/CountCommand.java [hunk 3]
```diff
@@ -16,17 +16,24 @@
...
                 StringBuilder message = new StringBuilder()
                     .append(ChatColor.GOLD)
                     .append("Current count: ")
                     .append(result.getTotal())
                     .append("/")
-                    .append(Utils.getMaxHeads(p))
+                    .append(Utils.getMaxHeads(player))
                     .append('\n');
 
+                if (Utils.canBypass(player)) {
```
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/headlimiter/CountCommand.java:30]: Does this still display how many heads are in a chunk.

**Review by J3fftw1 — COMMENTED** [review]
Mind testing the commands and screenshot what they display and test or get someone to test will approve after that ❤️

File: src/main/java/dev/j3fftw/headlimiter/CountCommand.java (hunk 3 above)
Comment (WalshyDev) [reply src/main/java/dev/j3fftw/headlimiter/CountCommand.java:30]: Yes

**Comment by WalshyDev** [comment]
standard:
image

bypass:
image
what color do we want that bypass text?

**Comment by J3fftw1** [comment]
> standard: image
> 
> bypass: image what color do we want that bypass text?

Could you make if its over limit make it red
Also move it above the count it shouldnt be in the sum up of what all are in the limit
Soo
Bypass blah blah
Current count
Cargo1
Cargo2
Cargo3

**Comment by WalshyDev** [comment]
no bypass:
image

bypass at limit:
image

bypass under limit:
image

**Review by J3fftw1 — CHANGES_REQUESTED** [review]
He held me at gun point and forced me to merge

**Review by J3fftw1 — APPROVED** [review]
Miss clicked

**Pull request merged**